dace/transformation/transformation.py

```diff
@@ -535,6 +537,39 @@ def apply(self, sdfg, *args, **kwargs):
         state.remove_node(node)
         type(self).postprocessing(sdfg, state, expansion)
 
+    def to_json(self, parent=None) -> Dict[str, Any]:
+        props = serialize.all_properties_to_json(self)
+        return {
+            'type': 'ExpandTransformation',
+            'transformation': type(self).__name__,
+            'classpath': nd.full_class_path(self),
+            **props
+        }
+
+    @staticmethod
+    def from_json(json_obj: Dict[str, Any],
+                  context: Dict[str, Any] = None) -> 'ExpandTransformation':
+        xform = pydoc.locate(json_obj['classpath'])
```

Comment:
what if it doesn't exist? does it become a serializableobject?

Reply:
Yes, exactly - it diverts to `Transformation.from_json`, which in turn fails and generates a SerializableObject